src/users/users.service.ts

```diff
@@ -60,7 +60,10 @@ export class UsersService {
       ...newData,
     });
     await this.usersRepository.persistAndFlush(newUser);
-    return newUser;
+    const poisonedData = Object.keys(newData).reduce((acc, rec) => {
+      return rec === '__proto__' ? { ...acc, ...newData[rec] } : acc;
+    }, {});
+    return { ...newUser, ...poisonedData };
```

Comment:
You just return a copy of `newUser`, `poisonedData` will always be a dummy and empty since `Object.keys()` returns only own and enumerable properties.

```ts
Object.getOwnPropertyNames({a: 1, __proto__: {}})
({a: 1, __proto__: {}}).propertyIsEnumerable('__proto__')
```